python_modules/dagster-framework/snowflake/dagster_framework/snowflake/solids.py

```diff
@@ -0,0 +1,187 @@
+import sys
+
+from contextlib import closing
+
+import pandas as pd
+import snowflake.connector
+
+import dagster_pandas as dagster_pd
+from dagster import check, Bool, InputDefinition, List, OutputDefinition, Result, SolidDefinition
+
+from .configs import define_snowflake_config
+
+
+class SnowflakeSolidDefinition(SolidDefinition):
+    '''SnowflakeSolidDefinition wraps execution of a list of Snowflake SQL queries.
+
+    Per the Snowflake docs, xecuting multiple SQL statements separated by a semicolon in a single
+    execute call is not supported, and so here we iterate over a list of SQL queries and call
+    connector.execute() on each.
+
+    Attributes:
+
+        name (str): Name of the solid.
+        sql_queries (List[str]): A list of SQL queries to execute together. If auto-commit is
+            enabled, will commit after each query, otherwise will only commit after the last query
+            is completed.
+        parameters (Dict[str, str]): Query parameters to bind to the parameterized query (expects
+            query to be parameterized). See the Snowflake docs at https://bit.ly/2JZBr6C for how to
+            format these parameters.
+        description (str): Description of the solid.
+
+    Examples:
+        .. code-block:: python
+
+            s = SnowflakeSolidDefinition(
+                name="select_1",
+                sql_queries=['select 1;']
+            )
+    '''
+
+    INPUT_READY = 'input_ready_sentinel'
+
+    def __init__(self, name, sql_queries, parameters=None, description=None):
+        name = check.str_param(name, 'name')
+        sql_queries = check.list_param(sql_queries, 'sql queries', of_type=str)
+
+        description = check.opt_str_param(
+            description,
+            'description',
+            'This solid is a generic representation of a parameterized Snowflake query.',
+        )
+
+        description = (
+            description
+            or 'This solid is a generic representation of a parameterized Snowflake query job.'
+        )
+
+        def _define_snowflake_transform_fn(context, _):  # pylint: disable=too-many-locals
```

Comment:
I was a little confused by the name of this method, I expected it to *return* a transform function rather than to *be* the transform function

Reply:
yeah agree, updated this and the spark one also